src/adapter/input/post_router.py

```
@@ -0,0 +1,18 @@
+from port.input.post_app_service import PostApplicationService
+
+router = APIRouter(prefix="/posts")
+
+def get_post_application_service():
+    db = SessionLocal()
+    adapter = PostReposiotryImpl(db)
+    usecase = PostUseCase(adapter)
+    return PostApplicationService(usecase)
+
+@router.post(path='', status_code=status.HTTP_201_CREATED)
+def create_post(post: Post, service: PostApplicationService = Depends(get_post_application_service)):
```

Ruff (0.8.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)